src/Server.cpp

```diff
@@ -41,12 +60,17 @@ void Server::startPollEventLoop()
 {
 	addServerSocketsPollFdToVectors();
 	int pollCounter = 0;
+	int timeout = -1;
 	while (1)
 	{
+		if (_hasCGI)
+			timeout = 500;
```

Comment:
Does it mean that CGI can be executed max 500ms?

Reply:
Nope. This means we wait 500 ms on poll and then we go back to waitpid, we pass through poll. We wait 10 long seconds for the CGI to complete. This is another variable or macro.